src/common/liboptparse/optparse.c

```diff
@@ -1167,9 +1167,19 @@ static struct option * option_table_create (optparse_t *p, char **sp)
     /*
      *  Initialize final element of option array to zeros:
      */
-    if (opts)
+    if (opts) {
         memset (&opts[j], 0, sizeof (struct option));
 
+        /* Corner case, no options configured by user */
+        if (sp && !(*sp)) {
```

Comment:
This is a good fix, Thanks! However, handling the case down here took me a minute to grok, and seems a bit out of place after the fact. Would it work if we initialized `*sp` with `optstring_create()` at the top of the function? Then it is just "proper initialization" and we don't have to handle no options as a corner case (though `optstring_create()` makes the string as `"+"` so we'd have to see if that is valid input to getopt_long...)